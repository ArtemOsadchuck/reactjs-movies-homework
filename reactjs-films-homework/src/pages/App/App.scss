$body-bg-color: rgb(21, 21, 21);

body {
  height: max-content;
  min-height: 100vh;

  .wrapper {
    background: $body-bg-color;
    font-family: "Roboto", sans-serif;
    height: max-content;
    margin: 0 auto;
    min-height: 100vh;
    width: 100vw;
  }
}
