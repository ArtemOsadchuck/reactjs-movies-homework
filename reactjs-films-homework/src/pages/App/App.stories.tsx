import React, { ComponentProps } from 'react';

import { Story } from '@storybook/react';
import App from './App';

export default {
  title: 'App',
  component: App,
};

const Template: Story<typeof App> = () => <App />;

export const FirstStory = Template.bind({});
