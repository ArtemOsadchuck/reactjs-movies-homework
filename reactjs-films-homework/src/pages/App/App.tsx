import React from 'react';

import Header from '../../components/Header/Header';
import Main from '../Main';
import MovieDetails from '../MovieDetails';
import ActorProfile from '../ActorProfile';
import ErrorBoundary from '../../components/ErrorBoundary';

import './App.scss';
import { Route } from 'react-router-dom';

function App() {
  return (
    <div className="wrapper">
      <ErrorBoundary title="header component">
        <Header />
      </ErrorBoundary>

      <ErrorBoundary title="main page">
        <Route exact path="/">
          <Main />
        </Route>
      </ErrorBoundary>

      <ErrorBoundary title="movie details page">
        <Route path="/movie-details/">
          <MovieDetails />
        </Route>
      </ErrorBoundary>

      <ErrorBoundary title="actor profile page">
        <Route path="/actor-profile">
          <ActorProfile />
        </Route>
      </ErrorBoundary>
    </div>
  );
}

export default App;
