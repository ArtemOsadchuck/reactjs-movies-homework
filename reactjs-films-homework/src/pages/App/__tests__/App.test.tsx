import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';

test('App test', () => {
  const { asFragment } = render(<App />);
  //   screen.debug();
  expect(asFragment).toMatchSnapshot();
});
