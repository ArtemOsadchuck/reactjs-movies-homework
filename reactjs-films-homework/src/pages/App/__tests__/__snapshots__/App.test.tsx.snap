// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="boundaryWrapper"
    >
      <h2
        color="wite"
      >
        An error has occurred on the header component. Please try to refresh the page
      </h2>
    </div>
    <div
      class="boundaryWrapper"
    >
      <h2
        color="wite"
      >
        An error has occurred on the main page. Please try to refresh the page
      </h2>
    </div>
    <div
      class="boundaryWrapper"
    >
      <h2
        color="wite"
      >
        An error has occurred on the movie details page. Please try to refresh the page
      </h2>
    </div>
    <div
      class="boundaryWrapper"
    >
      <h2
        color="wite"
      >
        An error has occurred on the actor profile page. Please try to refresh the page
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
